### DISCUSSION TITLE
Unable to push RubyGem to Package Registry (The expected resource was not found)

### DISCUSSION THREAD

**Discussion by Chew** [community/community#24748] [Apps, API and Webhooks]
<p>I’ve been able to push gems before but now for some reason I’m unable to? However, it was on my personal account and now I’m trying on an org account.</p>
<p>I have my credentails stored and saved</p>
<h2>$ cat ~/.gem/credentials</h2>
<p>:rubygems_api_key: [key]<br>
:github: Bearer [key]</p>
<p>But when I go to push it, using the following command:</p>
<p>gem push --key github --host [<a href="https://rubygems.pkg.github.com/" rel="nofollow noopener">https://rubygems.pkg.github.com/</a><a href="https://rubygems.pkg.github.com/%5BORG" rel="nofollow noopener">ORG</a>] [REPO]-0.1.0.gem</p>
<p>It fails, with this error:</p>
<p>Pushing gem to <a href="https://rubygems.pkg.github.com/Memerator" rel="nofollow noopener">https://rubygems.pkg.github.com/x</a>…<br>
The expected resource was not found.</p>
<p>What should I do? I’ve already made sure to select way more scopes than I probably need</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/d/d9543f908217a3a34b2f8db8e8596936afaaf7e4.png" data-download-href="/uploads/short-url/v0A6XqLuAA4hqttWxF4cOXvTzNy.png?dl=1" title="unknown" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/8278263/181086556-9de5199b-475d-49e7-a6cb-9e0e749d8f29.png" alt="unknown" data-base62-sha1="v0A6XqLuAA4hqttWxF4cOXvTzNy" width="690" height="298" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/d/d9543f908217a3a34b2f8db8e8596936afaaf7e4_2_690x298.png, https://aws1.discourse-cdn.com/github/optimized/2X/d/d9543f908217a3a34b2f8db8e8596936afaaf7e4_2_1035x447.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/d/d9543f908217a3a34b2f8db8e8596936afaaf7e4_2_1380x596.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/d/d9543f908217a3a34b2f8db8e8596936afaaf7e4_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">unknown</span><span class="informations">1518×656 81.9 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p> At first I thought it might be the dashes in the repo name, however the previous push (that worked) has dashes. </p>
<p>I’m out of ideas, what’s the situation here? We’re dealing with a 404 here, not a 401, so the error must have to do with some naming convention.</p>

**Comment by valedan** [comment]
<p>Did you ever figure this out? I’m having the exact same issue, things work fine for a personal repo but I get the same error trying to publish to an org repo.</p>

**Comment by sebastian-schuth** [comment]
<p>Same here - It works for one of our org-private projects (and with org-private packages), but with the same config, it fails for another.</p>
<p>Using an issuficient token raises a different error, so I suppose my token is ok.</p>

**Comment by sebastian-schuth** [comment, marked as answer]
<p>Solved it with the help of the GH support. or me, it was the gem name in the spec - it has to match the name of the repository within your organisazion.</p>
<p>So, for a repo</p>
<p>great-org/foobar-baz</p>
<p>, the gem needs to have the name</p>
<p>‘foobar-baz’</p>
<p>. Otherwise, GH packages seems to be unable to link back to your repo, giving you the somewhat misleading error message.</p>

**Reply by akasyagroup** [reply]
> GH desteği yardımıyla çözüldü. ya da ben, spesifikasyondaki mücevher adıydı - kuruluşunuzdaki veri havuzunun adıyla eşleşmesi gerekir.
> 
> Yani, bir repo için
> 
> harika-org/foobar-baz
> 
> , mücevherin adı olmalı
> 
> 'foobar-baz'
> 
> . Aksi takdirde, GH paketleri deponuza geri bağlanamıyor gibi görünüyor ve size biraz yanıltıcı hata mesajı veriyor.

**Comment by reiz** [comment]
<p>I had the same problem and + 1. For me the problem was an underscore in the repo name. As long the repo name was “ve_core” the push didn’t worked for me. After I renamed it to “ve-core” it worked fine. </p>

**Comment by mathieujobin** [comment]
<p>this is truly annoying.<br>
I can’t possibly rename all my gems because I want to host them on Github</p>
<p>that’s a huge fail<br>
I need to push my gems as-is.</p>

**Comment by ytaben** [comment]
<p>If someone stumbles on this:<br>
Add the following to your gemspec (replace OWNER and REPO_NAME):</p>
<pre><code class="lang-auto">spec.metadata = {
    'github_repo' =&gt; 'git@github.com:OWNER/REPO_NAME.git'
  }
</code></pre>

**Reply by luiscvega** [reply]
THIS

**Reply by mathieujobin** [reply]
Thank you

**Reply by jaimecgomezz** [reply]
Thanks!
